File: .guides/tests/ch-1.js

turnTaken  = null;

var fileName = window.location.origin + '/public/py/' + window.testEnv.cmd + '.py';

$.get(fileName)
.success(function(data) {

  if(data == 404) {
    console.log(exception);
    codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.INVALID, fileName + ' not found'); 
  }

  try {        
    createEmptyMaze();
    
    // load python
    var module = Sk.importMainWithBody("", false, data);

    // get reference to functions

    turnTaken = module.tp$getattr('turnTaken');
    
    // test

    var _well = false;
    var _ok = false;

    if(typeof turnTaken != 'undefined') {
      energy = 9;
      steps = 10;

      window.showMessage = function (val) {
        if(val.toLowerCase() == 'not going well') _well = true;
      }

      Sk.misceval.callsim(turnTaken);

      energy = 10;
      steps = 10;

      window.showMessage = function (val) {
        if(val.toLowerCase() == 'going ok') _ok = true;
      }

      Sk.misceval.callsim(turnTaken);

      if(!_well || !_ok) {
        return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
      }
    }
    else {
      return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
    }

    codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.SUCCESS, 'Well done!');    

  }
  catch(exception) {
    console.log('skulpt exception');
    console.log(exception);
    codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.INVALID, exception.toString()); 
  }
})
.fail(function(jqxhr, settings, exception) {
  console.log('jqxhr fail');
  console.log(exception);
  codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.INVALID, exception.message); 
});    

File: .guides/tests/ch-1.py

energy = 9
steps = 10
message = ''

def showMessage(val):
  global message
  message = val

def getEnergy():
  global energy
  return energy
  
def getSteps():
  global steps
  return steps

try:
  execfile('/home/codio/workspace/public/py/ch-1.py')
  
  turnTaken()
  if message.lower().strip() != 'not going well':
    raise ValueError("turnTaken is not showing the 'Not going well' message for the correct scenarios")

  energy = 10
  turnTaken()
  if message.lower().strip() != 'going ok':
    raise ValueError("turnTaken is not handling the scenario where energy is greater than equal to 10")
  
  energy = 9
  steps = 9
  turnTaken()
  if message.lower().strip() != 'going ok':
    raise ValueError("turnTaken is not handling the scenario where steps is less than 10")
  
  print 'well done'
  exit(0)
  
except (IOError, SyntaxError, ValueError) as e:
  print str(e)
  exit(1)
  
print 'Not quite right, try again!'
exit(1)
